src/optimizer/join_order/join_order_optimizer.cpp

```diff
@@ -10,19 +10,27 @@
 
 namespace duckdb {
 
-JoinOrderOptimizer::JoinOrderOptimizer(ClientContext &context) : context(context), query_graph_manager(context) {
+JoinOrderOptimizer::JoinOrderOptimizer(ClientContext &context)
+    : context(context), query_graph_manager(context), depth(1) {
 }
 
 JoinOrderOptimizer JoinOrderOptimizer::CreateChildOptimizer() {
 	JoinOrderOptimizer child_optimizer(context);
 	child_optimizer.materialized_cte_stats = materialized_cte_stats;
 	child_optimizer.delim_scan_stats = delim_scan_stats;
+	child_optimizer.depth = depth + 1;
 	return child_optimizer;
 }
 
 unique_ptr<LogicalOperator> JoinOrderOptimizer::Optimize(unique_ptr<LogicalOperator> plan,
                                                          optional_ptr<RelationStats> stats) {
 
+	if (depth > 25) {
```

Comment:
nit: this could be a static constexpr like `THRESHOLD_TO_SWAP_TO_APPROXIMATE` in plan_enumerator.cpp